src/auth.rs

```diff
@@ -61,30 +61,34 @@
 //! ```
 //! [`SecretKey`]: https://docs.rs/orion/latest/orion/auth/struct.SecretKey.html
 //! [`SecretKey::default()`]: https://docs.rs/orion/latest/orion/auth/struct.SecretKey.html
-//! [`Tag`]: https://docs.rs/orion/latest/orion/hazardous/mac/hmac/struct.Tag.html
+//! [`Tag`]: struct.Tag.html
 
-use crate::{errors::UnknownCryptoError, hazardous::mac::hmac};
-pub use crate::{hazardous::mac::hmac::Tag, hltypes::SecretKey};
+pub use crate::hltypes::{Blake2bTag as Tag, SecretKey};
+use crate::{errors::UnknownCryptoError, hazardous::hash::blake2b};
+
+/// The recommended Tag size (bytes) to be output by Blake2b in keyed mode.
+const BLAKE2B_TAG_SIZE: usize = 16;
 
 #[must_use]
-/// Authenticate a message using HMAC-SHA512.
+/// Authenticate a message using Blake2b in keyed mode.
 pub fn authenticate(secret_key: &SecretKey, data: &[u8]) -> Result<Tag, UnknownCryptoError> {
-	let mut state = hmac::init(&hmac::SecretKey::from_slice(
-		secret_key.unprotected_as_bytes(),
-	)?);
+	let blake2b_secret_key = blake2b::SecretKey::from_slice(secret_key.unprotected_as_bytes())?;
```

Comment:
I think it would be worthwhile to enforce a 32-byte secret key minimum size. Could you please add a check to make sure the `secret_key` is greater-than or equal to 32 in length? If it is not, `UnknownCryptoError` needs to be returned. By enforcing 32-byte minimum, we also don't need to specify the lower length limitation of the `hazardous` secret key, which cannot be 0 or empty.

This restriction should be mentioned in the errors-section of the docs and have an accompanying unit-test.